pkg/graveler/graveler.go

```diff
@@ -1745,6 +1779,38 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 	return newCommitID, nil
 }
 
+func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repositoryID RepositoryID, branchID BranchID, f BranchUpdateFunc) error {
+	const (
+		maxInterval = 5
+		setTries    = 3
+	)
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = maxInterval * time.Second
+
+	var try int
+	err := backoff.Retry(func() error {
+		// TODO(3586) - if the branch commit id hasn't changed, update the fields instead of fail
+		err := g.RefManager.BranchUpdate(ctx, repositoryID, branchID, f)
+		if err != nil {
+			if !errors.Is(err, kv.ErrPredicateFailed) {
+				return backoff.Permanent(err)
+			}
```

Comment:
1. For readability - we can move this check before err != nil
2. The try < setTries can be check with err != nil together

Reply:
done